server/src/main/java/org/opensearch/common/cache/stats/DummyCacheStatsHolder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache.stats;
+
+import org.opensearch.common.metrics.CounterMetric;
+
+import java.util.List;
+
+/**
+ * A dummy version of CacheStatsHolder, which cache implementations use when FeatureFlags.PLUGGABLE_CACHES is false.
+ * Will be removed once pluggable caches is no longer an experimental feature.
+ * Returns all-zero stats when calling getImmutableCacheStatsHolder(). Does keep track of entries for use in ICache.count().
+ */
+public class DummyCacheStatsHolder implements CacheStatsHolderInterface {
+    private CounterMetric entries;
+    private CounterMetric sizeInBytes;
+    private final List<String> dimensionNames;
+
+    public DummyCacheStatsHolder(List<String> dimensionNames) {
+        this.dimensionNames = dimensionNames;
+        this.entries = new CounterMetric();
+        this.sizeInBytes = new CounterMetric();
+    }
+
+    @Override
+    public void incrementHits(List<String> dimensionValues) {}
+
+    @Override
+    public void incrementMisses(List<String> dimensionValues) {}
+
+    @Override
+    public void incrementEvictions(List<String> dimensionValues) {}
+
+    @Override
+    public void incrementSizeInBytes(List<String> dimensionValues, long amountBytes) {
+        sizeInBytes.inc(amountBytes);
```

Comment:
Why do we have even do this? All the methods should ideally do nothing as anyway we need to return zero values for all such metrics? 
Plus no need to pass dimensionNames or keep CounterMetric related classes.

Reply:
Oops - I had added this counter for sizeInBytes before I realized we couldn't track it in the no-op stats holder, because we would have to add a sizeInBytes() function in ICache for it to be accessible without also having a non-zero size reported in the API. Will remove. 

However we do have to keep the entries CounterMetric so that count() can work properly.

Reply:
Why do we still need CounterMetric in NoopStatsHolder?
Are u referring to count() in IRC, but that will still work without stats as well?